quaternions/utils.py

```diff
@@ -0,0 +1,38 @@
+import numpy as np
+
+
+def covariance_matrix_from_angles(angles_list):
+    """
+    Computes covariance matrix from 3d vectors list
+    :param angles_list: list of NumPy arrays
+    """
+    covariance_matrix = np.zeros((3, 3))
+    for angles in angles_list:
+        covariance_matrix += np.outer(angles.T, angles)
+    covariance_matrix /= (angles_list.shape[0] + 1)
+    return covariance_matrix
+
+
+def sigma_lerner(covariance_matrix):
+    """ Computes sigma lerner from covariance matrix"""
+    values, _ = np.linalg.eig(covariance_matrix)
+    return 1.87 * np.sqrt(max(values))
+
+
+def rotation_matrix(quaternion):
```

Comment:
this is not a rotation matrix, but the matrix of a cross product with the vector `[qi, qj, qk]`.
The name `cross_product_matrix` might be better. It could also have the same name it has in the paper where it comes from.

Reply:
done